demisto_sdk/commands/download/downloader.py

```diff
@@ -220,14 +220,38 @@ def fetch_custom_content(self) -> bool:
             for member in tar.getmembers():
                 file_name: str = self.update_file_prefix(member.name.strip('/'))
                 file_path: str = os.path.join(self.custom_content_temp_dir, file_name)
-                with open(file_path, 'w') as file:
-                    extracted_file = tar.extractfile(member)
-                    # File might empty
-                    if extracted_file:
-                        string_to_write = extracted_file.read().decode('utf-8')
+
+                extracted_file = tar.extractfile(member)
+                # File might empty
+                if extracted_file:
+                    string_to_write = extracted_file.read().decode('utf-8')
+
+                    is_playbook_downloaded = False
+
+                    if not self.list_files and re.search(r'playbook-.*\.yml', member.name):
+                        # if playbook and not list-file flag, we should download the file via direct REST API
+                        # because there are props like scriptName, that playbook from custom content bundle don't contain
+
+                        file_yaml_object = yaml.load(string_to_write)
+                        playbook_id = file_yaml_object.get('id')
+                        playbook_name = file_yaml_object.get('name')
+                        if playbook_id and playbook_name:
+                            if playbook_name not in self.input_files:
```

Comment:
when self.all_custom_content flag is True, self.input_files at this stage still doesn't contain the playbook name (only after self.handle_all_custom_content_flag()) so we can't rely on it. Let's make sure handle_all_custom_content_flag is called before this function.